python/ray/_private/utils.py

```diff
@@ -1197,6 +1197,58 @@ def check_dashboard_dependencies_installed() -> bool:
         return False
 
 
+def internal_kv_get_with_retry(gcs_client, key, namespace, num_retries=20):
+    result = None
+    if isinstance(key, str):
+        key = key.encode()
+    for _ in range(num_retries):
+        try:
+            result = gcs_client.internal_kv_get(key, namespace)
+        except Exception:
+            logger.exception("Internal KV Get failed")
+            result = None
+
+        if result is not None:
+            break
+        else:
+            logger.debug(f"Fetched {key}=None from redis. Retrying.")
+            time.sleep(2)
+    if not result:
+        raise RuntimeError(
+            f"Could not read '{key}' from GCS (redis). "
+            "If using Redis, did Redis start successfully?"
+        )
+    return result
+
+
+def internal_kv_put_with_retry(gcs_client, key, value, namespace, num_retries=20):
+    if isinstance(key, str):
+        key = key.encode()
+    error = None
+    for _ in range(num_retries):
+        try:
+            return gcs_client.internal_kv_put(
+                key, value, overwrite=True, namespace=namespace
+            )
+        except grpc.RpcError as e:
```

Comment:
why are we catching RpcError here vs Exception in internal_kv_get_with_retry

Reply:
Hmm I am not sure about the reasoning actually. This part is just refactoring. I am planning to move this util method into gcs_client and add num_retires instead, so we can talk about more details in that PR.